### DISCUSSION TITLE
https:// and wss:// möglich?

### DISCUSSION THREAD

**Discussion by TomRadfahrer** [tbnobody/OpenDTU#355] [Q&A] [answered]
Die openDTU hat in meinem lokalen Netzwerk im Moment die IP-Adresse 192.168.0.54. Mit der Adresse "http://192.168.0.54" kann ich in meinem lokalen Netzwerk openDTU erreichen, mit der Adresse "https://192.168.0.54" aber nicht. 

Das gleiche gilt leider auch für den Zugriff auf die WebSocket-Schnittstelle, ws://192.168.0.54 geht, aber wss://192.168.0.54 geht nicht. Und auf die WebSocket-Schnittstelle möchte ich direkt über eine Apple-App zugreifen. Jetzt legt mir die App bei dem ungesichertem Zugriff auf ws:// aber Steine in den Weg, Apple ist der Meinung, dass man heutzutage doch die verschlüsselten Protokolle verwenden soll, die unverschlüsselten Protokolle haben heute Daseinsberechtigung mehr.

Was muss ich tun, damit openDTU über HTTPS und WSS ansprechbar ist? Ist das einfach möglich?  Im voraus vielen Dank.

P.S. Dies ist kein Feature Request. Wenn ich lediglich beim Compilieren einen Schalter setzen muss, schön. Wenn es nicht so einfach geht, programmiere ich in meiner App eben eine Exception für das lokale Netzwerk.

**Comment by tbnobody** [comment, marked as answer]
SSL ist auf solch kleinen Geräten wie ESP's die Hölle und sehr Resourcenintensiv.

**Reply by TomRadfahrer** [reply]
Danke für die schnelle Antwort, verstanden. Dann werde ich Exceptions für meine App definieren.

**Reply by stefan123t** [reply]
@tbnobody ich habe es irgendwo schon mal mit der BearSSL Bibliothek von earlphilhower zum Laufen bekommen. Nur alle Chain Certificates zu prüfen ist PITA: ich weiß nicht mehr ob ich mich auf das LetsEncrypt Root CA oder mein Server Certificate beschränkt habe. Wenn es Dich interessiert suche ich den Code für den ESP8266 noch mal raus ...

**Reply by scorpi11** [reply]
Ein zwischengeschalteter SSL-Proxy im LAN (geht z.B. mit lighttpd oder nginx) sollte auch klappen, nur ist die App dann natürlich entsprechend eingeschränkt nutzbar.
